proposed/vcpkg.md

```diff
@@ -0,0 +1,91 @@
+- @mmitche
+- @jkotas
+- @AaronRobinsonMSFT
+- @wtgodbe
+
+## Design
+
+### Acquiring vcpkg
+
+Arcade should provide functionality in the scripts in `eng/common` to acquire the vcpkg tool.
+vcpkg will be able requested by an entry in the global.json file for a repository.
+The Arcade build scripts will define which version of vcpkg is used.
+
+As we will be using our own custom registry, Arcade can do a sparse checkout of the vcpkg repository to reduce the amount of data that needs to be downloaded, excluding the default registry.
+
+To support offline builds, the vcpkg repo and a matching version of the [vcpkg-tool repository](https://github.com/microsoft/vcpkg-tool) will be included as a submodule in the source-build-externals repository.
+
+### Custom registry
+
+We can either create a new repository for the registry or use the source-build-externals repository as a vcpkg registry as well. This registry will, similar to the `dotnet-public` NuGet feed, contain all of the public packages that we want to consume from within our projects. However, unlike the `dotnet-public` NuGet feed, we will need to modify the portfiles for the packages that we want to consume to not use online sources in an offline build. This likely means that for each package, we'll need to create a submodule in source-build-externals or another repository for each package that we want to consume.
+
+Alternatively, we could build an alternative mechanism in the VMR that pre-clones all dependencies when updating the VMR. This option seems more expensive, and as such we should avoid it if possible.
+
+### Using vcpkg
+
+All repositories that use vcpkg must use the manifest mode of vcpkg to declare their dependencies. Classic mode is not supported.
+
+### Source-build partners and conditionally system dependencies
+
+For some dependencies, we may want to use a system-installed version of the dependency in source-build scenarios.
+For these cases, our source-build partners can remove the dependencies from the vcpkg manifest.
+Since the CMake integration with vcpkg uses the standard `find_library` and `find_package` CMake commands, the system dependencies will be found instead of the vcpkg dependencies with no change to the CMake source code.
+
+### Flowing native dependencies with binary caching
+
+vcpkg has a binary caching feature that can use NuGet.
+We should be able to utilize this feature for flowing native dependencies that are consumed by native code (such as `dotnet/llvm-project` assets) to avoid having to run the build through MSBuild to locate the dependencies or rebuilding the dependencies in each repository. This is not something that needs to be done immediately, but is something we can consider in the future instead of our existing package flow model.
+
+## Q & A
+
+- Why use vcpkg instead of another package manager?
+    - Vcpkg is already used by many teams at Microsoft, and is already used by many of the dependencies that we may want to consume.
```

Comment:
I understand that vcpkg is useful for applications.
 
What are the open source projects similar to .NET runtime that use vcpkg? For example, are there open source projects that use vcpkg and that are included in mainstream Linux distros? How do they deal with the offline build requirements, etc.?

Reply:
I'm not sure if there are any projects that use vcpkg and are included in mainstream Linux distros. To my knowledge, most projects with non-test dependencies that are included in mainstream Linux distros have their build-time dependencies available as dependencies in the distro (through -dev packages or the source packages) and they reference them as shared libraries at runtime. Testing dependencies are usually vendored in (LLVM vendors GoogleTest and Google Benchmark).

We may be the first in this space.

The offline scenario is why I was thinking of vcpkg over another package manager. Using vcpkg allows us to use the same CMake whether the package is a system-provided dependency or a vcpkg-provided dependency, enabling our Linux distro partners to easily switch dependencies to system-provided if desired.